server/server.go

```diff
@@ -352,12 +336,34 @@ func configureStorage(conf *Config) (storageProvider storage.Store, err error) {
 	}
 
 	if storageProvider == nil {
-		storageProvider, err = util.NewSqliteStorage(":memory:")
+		storageProvider, err = util.NewSqliteStorage(sqlite.InMemory)
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	if conf.ChainID == flowgo.Testnet || conf.ChainID == flowgo.Mainnet {
+		// TODO: any reason redis shouldn't work?
+		baseProvider, ok := storageProvider.(*sqlite.Store)
+		if !ok {
+			return nil, fmt.Errorf("only sqlite is supported with forked networks")
+		}
```

Comment:
I think we can wrap any storage provider there. Tricky part is snapshot and persist there.